### PR TITLE
Fix AmigoCloud driver on Windows

### PR DESCRIPTION
**Pull request by vchernetsky**
On Windows environments (tested through Command Line and QGIS on
Windows) the environment variables were not passed correctly.
Switch to having an option to receive an AmigoCloud API key in
the connection string.

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp [hunk 1]
```diff
@@ -162,6 +162,11 @@ int OGRAmigoCloudDataSource::Open( const char * pszFilename,
     osAPIKey = CSLFetchNameValueDef(papszOpenOptionsIn, "API_KEY",
                                     CPLGetConfigOption("AMIGOCLOUD_API_KEY", ""));
 
+	if (osAPIKey.empty())
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp]: Please use 4 space indentation

File: gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp [hunk 2]
```diff
@@ -162,6 +162,11 @@ int OGRAmigoCloudDataSource::Open( const char * pszFilename,
...
+	{
+		osAPIKey = OGRAMIGOCLOUDGetOptionValue(pszFilename, "API_KEY");
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp]: Should be documented in the html page

**Comment by rouault** [comment]
What is the issue exactly with environment variable on Windows ? That's surprising

**Comment by rouault** [comment]
@vchernetsky Feedback appreciated quickly if you want this to be part of GDAL 2.1.2 that is going to be cut this friday.

**Comment by vchernetsky** [comment]
@rouault I have fixed the issues you have pointed out, merged OSGeo:trunk, and resolved merging conflicts. I see checks are failing, but those fails don't seem to be related to my changes. Is there anything I can do to get it moving along? Thanks!

**Comment by rouault** [comment]
Any reason not to keep the AMIGOCLOUD_PAGE_SIZE config option ? What is the issue exactly with  environment variable on Windows ?
Yes the failures are due to your branch being based on a broken trunk. Has been fixed since

**Comment by vchernetsky** [comment]
We removed AMIGOCLOUD_PAGE_SIZE parameter because it cannot be any other than 100, otherwise APIs don't work correctly. Environment variables on Windows probably fine, it was my error. Anyway, we pass those parameters through command line options, or API. There is no need for env. variables.

**Comment by rouault** [comment]
Committed in trunk r36639 and branches/2.1 r36640

**Comment by DanielCaldwell** [comment]
Hi @rouault,  we're not seeing these fixes in 2.1.3. I checked the 2.1 branch or the 2.1.3 tag and don't see it there. I do see it in trunk. I thought that it was merged to the 2.1 branches (per your comment above) and we were expecting it to go out in that release. Do you know if it was merged or is there something we need to do to get it merged? Thanks.

**Comment by rouault** [comment]
They went in the 2.1 branch per https://trac.osgeo.org/gdal/changeset/36640 and tag 2.1.3 was created from that branch later in https://trac.osgeo.org/gdal/log/tags/2.1.3/gdal/ogr/ogrsf_frmts/amigocloud/ , so they should be in 2.1.3

**Comment by vchernetsky** [comment]
The code 2.1.3 does not have my changes. For example line 134 of amigocloud/ogr_amigocloud.h:

134:	        int                         GetFeaturesToFetch() { return atoi(CPLGetConfigOption("AMIGOCLOUD_PAGE_SIZE", "500")); }

Should be:
136:        static  int                 GetFeaturesToFetch() { return 100; }

**Comment by rouault** [comment]
> The code 2.1.3 does not have my changes. For example line 134 of amigocloud/ogr_amigocloud.h:
> 
> 134: int GetFeaturesToFetch() { return atoi(CPLGetConfigOption("AMIGOCLOUD_PAGE_SIZE", "500")); }
> 
> Should be:
> 136: static int GetFeaturesToFetch() { return 100; }

Ah, indeed there must have been an issue in the backport from trunk to 2.1 branch and the change in ogr_amigocloud.h wasn't backported.
I just fixed that with r37545 "Amigocloud: backport part of change of trunk r36639 that should have been backported in 2.1 in r36640 but was missed for some reason (relates to https://github.com/OSGeo/gdal/pull/137)"

**Comment by vchernetsky** [comment]
Would it be possible to fix it, and make another 2.1 release soon?

**Comment by rouault** [comment]
I'd note that there's a period of a few days for testing the release candidate of each version, so it is obvious it wasn't used to do so. And there's no regression tests for that driver, so another reason why the issue wasn't noticed.

**Comment by vchernetsky** [comment]
The change wasn't just that one line. This whole pull request should be merged.
https://github.com/OSGeo/gdal/pull/137/files

**Comment by rouault** [comment]
The other files were already backported in 2.1.3 (see https://trac.osgeo.org/gdal/changeset/36640). Only the change in ogr_amigocloud.h was missed. Probably a conflict during the merge from trunk to 2.1 that went unnoticed.

**Comment by vchernetsky** [comment]
Oh, cool. Thanks! When could we expect next release?

**Pull request closed without merging**